bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/JsonParser.java

```diff
@@ -55,12 +47,10 @@
 @SuppressWarnings("unchecked")
 public class JsonParser {
 
-    private static ThreadLocal<StateMachine> tlStateMachine = new ThreadLocal<StateMachine>() {
-        @Override
-        public StateMachine initialValue() {
-            return new StateMachine();
-        }
-    };
```

Comment:
Can we remove JsonParser class?

Reply:
Some of the APIs use `NonStringValueProcessingMode` that does not have implementations in the `StreamParser`. I will add the implementation and remove the `JsonParser` altogether.